sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/implementation/AesKeyCryptographyClient.java

```diff
@@ -20,399 +18,367 @@
 import com.azure.security.keyvault.keys.models.JsonWebKey;
 import reactor.core.publisher.Mono;
 
+import javax.crypto.BadPaddingException;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
 import java.security.GeneralSecurityException;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
 import java.security.NoSuchAlgorithmException;
 import java.security.SecureRandom;
 import java.util.Objects;
 
 class AesKeyCryptographyClient extends LocalKeyCryptographyClient {
-    private static final ClientLogger LOGGER = new ClientLogger(AesKeyCryptographyClient.class);
-
-    private byte[] key;
+    private final byte[] aesKey;
 
     static final int AES_BLOCK_SIZE = 16;
 
-    AesKeyCryptographyClient(JsonWebKey key, CryptographyClientImpl implClient) {
-        super(implClient);
-        this.key = key.toAes().getEncoded();
-    }
+    AesKeyCryptographyClient(JsonWebKey jsonWebKey, CryptographyClientImpl implClient) {
+        super(jsonWebKey, implClient);
+
+        aesKey = jsonWebKey.toAes().getEncoded();
 
-    private byte[] getKey(JsonWebKey key) {
-        if (this.key == null) {
-            this.key = key.toAes().getEncoded();
+        if (aesKey == null || aesKey.length == 0) {
+            throw new IllegalArgumentException("The provided JSON Web Key cannot be null or empty.");
         }
-        return this.key;
     }
 
-
     @Override
-    public Mono<EncryptResult> encryptAsync(EncryptionAlgorithm algorithm,
-        byte[] plaintext,
-        JsonWebKey jsonWebKey,
-        Context context) {
+    public Mono<EncryptResult> encryptAsync(EncryptionAlgorithm algorithm, byte[] plaintext, Context context) {
         if (algorithm == null) {
-            return Mono.error(new NullPointerException("Encryption algorithm cannot be null."));
+            throw new NullPointerException("Encryption algorithm cannot be null.");
```

Comment:
Why did we change from returning Mono.error to throwing an exception? This method has async return type which should return Mono.error instead of throwing.

Reply:
If we return `Mono.error` we won't be able to log the exception at all. I don't think we should add a logger to `AesKeyCryptographyClient` or any other local client implementation and use `CryptographyAsyncClient`'s instead, which is what happens in the code that calls these methods such as:
```java
@ServiceMethod(returns = ReturnType.SINGLE)
public Mono<EncryptResult> encrypt(EncryptionAlgorithm algorithm, byte[] plaintext) {
    try {
        return withContext(context -> isLocalClientAvailable().flatMap(available -> {
            if (available) {
                verifyKeyPermissions(jsonWebKey, KeyOperation.ENCRYPT);

                return localKeyCryptographyClient.encryptAsync(algorithm, plaintext, context);
            } else {
                return implClient.encryptAsync(algorithm, plaintext, context);
            }
        }));
    } catch (RuntimeException ex) {
        return monoError(LOGGER, ex);
    }
}
```